nexus/client.go

```diff
@@ -0,0 +1,543 @@
+		}
+		return &StartOperationResult{
+			Pending: &OperationHandle{
+				Operation: options.Operation,
+				ID:        info.ID,
+				client:    c,
+			},
+		}, nil
+	case statusOperationFailed:
+		state, err := getUnsuccessfulStateFromHeader(response, body)
+		if err != nil {
+			return nil, err
+		}
+
+		failure, err := failureFromResponse(response, body)
+		if err != nil {
+			return nil, err
+		}
+
+		return nil, &UnsuccessfulOperationError{
+			State:   state,
+			Failure: failure,
+		}
+	default:
+		return nil, newUnexpectedResponseError(fmt.Sprintf("unexpected response status: %q", response.Status), response, body)
+	}
+}
+
+// ExecuteOperationOptions is input for [Client.ExecuteOperation].
+type ExecuteOperationOptions struct {
+	// Name of the operation to start.
+	Operation string
+	// Callback URL to provide to the handle for receiving async operation completions. Optional.
+	// Even though Client.ExecuteOperation waits for operation completion, some application may want to set this
+	// callback as a fallback mechanism.
+	CallbackURL string
+	// Request ID that may be used by the server handler to dedupe this start request.
+	// By default a v4 UUID will be generated by the client.
+	RequestID string
+	// Body of the operation request.
+	// If it is an [io.Closer], the body is guaranteed to be closed in Client.ExecuteOperation.
+	Body io.Reader
+	// Header to attach to start and get-result HTTP requests. Optional.
+	// Content-Type will be deleted in the get-result request.
+	Header http.Header
+	// Duration to wait for operation completion.
+	//
+	// ⚠ NOTE: unlike GetOperationResultOptions.Wait, zero and negative values are considered durations of MaxInt64.
+	Wait time.Duration
+}
+
+// NewExecuteOperationOptions is shorthand for creating an [ExecuteOperationOptions] struct with a JSON body. Marshals
+// the provided value to JSON using [json.Marshal] and sets the proper Content-Type header.
+func NewExecuteOperationOptions(operation string, v any) (options ExecuteOperationOptions, err error) {
+	if operation == "" {
+		err = errEmptyOperationName
+		return
+	}
+	var b []byte
+	b, err = json.Marshal(v)
+	if err != nil {
+		return
+	}
+	options.Operation = operation
+	options.Header = http.Header{headerContentType: []string{contentTypeJSON}}
+	options.Body = bytes.NewReader(b)
+	return
+}
+
+func (o *ExecuteOperationOptions) intoStartOptions() StartOperationOptions {
+	return StartOperationOptions{
+		Operation:   o.Operation,
+		CallbackURL: o.CallbackURL,
+		RequestID:   o.RequestID,
+		Header:      o.Header,
+		Body:        o.Body,
+	}
+}
+
+func (o *ExecuteOperationOptions) intoGetResultOptions() (options GetOperationResultOptions) {
+	options.Header = o.Header
+	if options.Header != nil {
+		options.Header = options.Header.Clone()
+		options.Header.Del(headerContentType)
+	}
+	if o.Wait <= 0 {
+		options.Wait = time.Duration(math.MaxInt64)
```

Comment:
I still think you should define infinite/max better. That a duration can only be a signed 64 bit integer is a Go-ism.

Reply:
`golang.org/x/time/rate` defines infinite duration like this:

```go
// InfDuration is the duration returned by Delay when a Reservation is not OK.
const InfDuration = time.Duration(math.MaxInt64)
```

And this is how they use it:
```go
	// Determine wait limit
	waitLimit := InfDuration
	if deadline, ok := ctx.Deadline(); ok {
		waitLimit = deadline.Sub(t)
	}
```